orchestrator/migrations/helpers.py

```diff
@@ -76,6 +76,177 @@ def create_missing_modify_note_workflows(conn):
+                    "product_type": "ProductType1",
+                    "description": "Product description",
+                    "tag": "ProductType",
+                    "status": "active",
+                    "product_block_ids": [
+                        "37afe017-5a04-4d87-96b0-b8f88a328d7a"
+                    ]
+                }
+            }
+    """
+    uuids = {}
+    for name, product in new.items():
+        product["name"] = name
+        product["product_id"] = str(product.get("product_id", uuid4()))
+        current_uuid = product["product_id"]
+        uuids[name] = current_uuid
+        conn.execute(
+            sa.text(
+                "INSERT INTO products (product_id, name, description, product_type, tag, status, created_at) "
+                "VALUES (:product_id, :name, :description, :product_type, :tag, :status, now())"
+            ),
+            product,
+        )
+        if "product_block_ids" in product:
+            for product_block_uuid in product["product_block_ids"]:
+                # Link many-to-many if product blocks are given.
+                    conn.execute(
+                        sa.text("INSERT INTO product_product_blocks VALUES (:product_id, :product_block_id)"),
+                        {
+                            "product_id": current_uuid,
+                            "product_block_id": product_block_uuid,
+                        },
+                    )
+        if "fixed_inputs" in product:
+            create_fixed_inputs(conn, current_uuid, product["fixed_inputs"])
+    return uuids
+
+
+def create_product_blocks(conn, new):
+    """
+    Create a new workflow
+
+    Args:
+        conn: DB connection as available in migration main file
+        new: an dict of your workflow data
+        products: list of product block ids to link these product blocks to
+    
+    Example:
+        >>> new = {
+                "Example Product Block": {
+                    "product_block_id": "37afe017-5a04-4d87-96b0-b8f88a328d7a",
+                    "description": "Product description",
+                    "tag": "ProductType",
+                    "status": "active",
+                },
+                "Example Product Block Two": {
+                    "product_type": "ProductBlock2",
+                    "description": "Product description",
+                    "tag": "ProductType",
+                    "status": "active",
+                }
+            }
+    """
+    uuids = {}
+    for name, product_block in new.items():
+        product_block["name"] = name
+        product_block["product_block_id"] = str(product_block.get("product_block_id", uuid4()))
+        uuids[name] = product_block["product_block_id"]
+        conn.execute(
+            sa.text(
+                "INSERT INTO product_blocks (product_block_id, name, description, tag, status, created_at) "
+                "VALUES (:product_block_id, :name, :description, :tag, :status, now())"
```

Comment:
You maybe want to add `ON CONFLICT DO NOTHING` to make sure that if the product block already exists it wont fail everything

Reply:
@howderek If you add @hmvp changes then Ill accept, thanks for this!

Reply:
@hmvp I have added the `ON CONFLICT DO NOTHING`, thanks